File: src/java/software/hsharp/api/icommon/IDatabase.java
package software.hsharp.api.icommon;

import javax.sql.DataSource;
import java.sql.SQLException;

public interface IDatabase {

	/**
	 * 	Create DataSource
	 *	@return data source
	 */
	public DataSource getDataSource();

    /**
     * 	Setup DataSource
     *	@param parameters connection pool parameters
     */
	public void setup(IDatabaseSetup parameters);

    /**
     * 	Connect
     *	@param connection connection
     */
	public void connect(ICConnection connection);

    /**
     * 	Get Status
     * 	@return status info or null if no local datasource available
     */
    public String getStatus();

    /**
     *  Get and register Database Driver
     *  @return Driver
     */
    public java.sql.Driver getDriver() throws SQLException;
}

File: src/kotlin/software/hsharp/api/icommon/IDatabase.kt
package software.hsharp.api.icommon

import javax.sql.DataSource

interface IDatabase {

    /**
     * Create DataSource
     * @return data source
     */
    val dataSource: DataSource

    /**
     * Get Status
     * @return status info or null if no local datasource available
     */
    val status: String

    /**
     * Get and register Database Driver
     * @return Driver
     */
    val driver: java.sql.Driver

    /**
     * Setup DataSource
     * @param parameters connection pool parameters
     */
    fun setup(parameters: IDatabaseSetup)

    val defaultSetupParameters : IDatabaseSetup

    /**
     * Connect
     * @param connection connection
     */
    fun connect(connection: ICConnection)
}
